src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1789,6 +1790,28 @@ public IndexDefinition GetIndexDefinition()
 
                             var retriever = GetQueryResultRetriever(query, documentsContext, fieldsToFetch, includeDocumentsCommand);
 
+                            if (query.Metadata.WhereMethods.TryGetValue(QueryMetadata.WhereMethod.RightSideMethod, out var method))
+                            {
+                                try
+                                {
+                                    if (query.Metadata.Query.Where is BinaryExpression be && be.Right is MethodExpression me)
+                                    {
+                                        var value = method.EvaluateSingleMethod((QueryResultRetrieverBase)retriever, null);
+                                        query.Metadata.WhereMethods.Remove(QueryMetadata.WhereMethod.RightSideMethod);
```

Comment:
note that QueryMetadata is cached and used concurrently

Reply:
This is important, we can't modify it